telemetry/metrics/init.go

```diff
@@ -0,0 +1,113 @@
+package metrics
+
+import (
+	"context"
+	"math/rand"
+
+	"github.com/gnolang/gno/telemetry/exporter"
+	"github.com/gnolang/gno/telemetry/options"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/metric"
+	sdkMetric "go.opentelemetry.io/otel/sdk/metric"
+)
+
+var (
+	ctx context.Context
+
+	// Metrics.
+	BroadcastTxTimer Int64Collector
+	BuildBlockTimer  Int64Collector
+)
+
+func Init(setCtx context.Context, config options.Config) error {
+	if config.ExporterEndpoint == "" {
+		return exporter.ErrEndpointNotSet
+	}
+
+	ctx = setCtx
```

Comment:
The meters and collectors are only initialized once at startup. The context passed into `Init` needs to be a different name than the package level variable `ctx`.  This is just initializing the context that can be used for all metrics since they are collected synchronously and we don't really have another context to work with at the time of collection.

Reply:
Having global package variable is highly discouraged for many reasons. also assigning/reassigning a value having package scope within the scope of a function and then reuse it somewhere else, it leads to unexpected side effect. Moreover makes hard to track down the usage of that.

Define the `context` and pass it to any underlying function when it is needed.

Reply:
Removed it